libs/feature/catalog/src/lib/my-org/my-org.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core'
+import { OrganizationsServiceInterface } from '@geonetwork-ui/common/domain/organizations.service.interface'
+import { AuthService } from '@geonetwork-ui/api/repository/gn4'
+import { BehaviorSubject, Observable } from 'rxjs'
+import { UserApiModel } from '@geonetwork-ui/data-access/gn4'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MyOrgService {
+  myOrgData$: Observable<{
+    orgName: string
+    logoUrl: string
+    recordCount: number
+    userCount: number
+    userList: UserApiModel[]
+  }>
+
+  private myOrgDataSubject = new BehaviorSubject<{
+    orgName: string
+    logoUrl: string
+    recordCount: number
+    userCount: number
+    userList: UserApiModel[]
+  }>({
+    orgName: '',
+    logoUrl: '',
+    recordCount: 0,
+    userCount: 0,
+    userList: [],
+  })
+
+  constructor(
+    private authService: AuthService,
+    private orgService: OrganizationsServiceInterface
+  ) {
+    this.myOrgData$ = this.myOrgDataSubject.asObservable()
+
+    this.authService.user$.subscribe((user) => {
+      const orgName = user?.organisation
+      // const orgName = 'Géo2France'
+      this.myOrgDataSubject.next({ ...this.myOrgDataSubject.value, orgName })
+    })
+
+    this.orgService.organisations$.subscribe((orgs) => {
+      const orgName = this.myOrgDataSubject.value.orgName
+      const org = orgs.find((org) => org.name === orgName)
+      const logoUrl = org?.logoUrl?.href.toString()
+      const recordCount = org?.recordCount
+      this.myOrgDataSubject.next({
+        ...this.myOrgDataSubject.value,
+        logoUrl,
+        recordCount,
+      })
+    })
+
+    this.authService.allUsers$.subscribe((users) => {
+      const orgName = this.myOrgDataSubject.value.orgName
+      // const orgName = 'Barbie Inc.'
+      const userList = users.filter((user) => user.organisation === orgName)
+      const userCount = userList.length
+      this.myOrgDataSubject.next({ ...this.myOrgDataSubject.value, userList })
+      this.myOrgDataSubject.next({ ...this.myOrgDataSubject.value, userCount })
+    })
```

Comment:
I am skeptical for this. I may need @jahow advice there ... 

Instead of subscribing to 3 observables, it may be good to do it with rxjs with some kind of (with combineLatest or zip, or something else if it exists.) : 
```typescript
const orgData$ = combineLatest([users$, allUsers$, organisations]).pipe(
  map(([a,b,c]) => resolve resolve),
);
```

and in html using an `async` pipe to get datas.

Reply:
Got it, thanks for the extensive explanation :slightly_smiling_face: 
Following this guidance, should I change the way `my-org-records.component.html` receives its data? 
Go from : `orgData?.logoUrl` to something like `myOrgRecordsService.myOrgData$.logoUrl | async` ?